libraries/AC_Avoidance/AP_OABendyRuler.cpp

```diff
@@ -112,13 +112,33 @@ bool AP_OABendyRuler::update(const Location& current_loc, const Location& destin
                     // calculate minimum margin to fence and obstacles for this scenario
                     float margin2 = calc_avoidance_margin(test_loc, test_loc2);
                     if (margin2 > _margin_max) {
+                        // if the chosen direction is directly towards the destination avoidace can be turned off 
+                        const bool active = (i != 0 || j != 0);
+                 
+                        float final_bearing = bearing_test;
+                        float final_margin = margin;
+                                                
+                        //check if we need to resist the change in direction of the vehicle. If we have a clear path to destination, go there anyhow  
+                        if(active && !is_zero(_bendy_ratio)) { 
```

Comment:
maybe is_positive(_bendy_ratio) to protect against negative values

Reply:
I am already checking for that in AP_OABendyRuler.h while importing the value from OAPathPlanner.cpp